tests/integration/nmstatectl_test.py

```diff
@@ -118,8 +118,12 @@ def test_set_command_with_yaml_format():
     ret = libcmd.exec_cmd(SET_CMD, stdin=ETH1_YAML_CONFIG)
     rc, out, err = ret
 
-    assert_rc(rc, RC_SUCCESS, ret)
+    assert_rc(rc, RC_SUCCESS, exec_cmd_report(ret))
 
 
-def assert_rc(actual, expected, return_tuple):
-    assert actual == expected, 'rc={}, out={}, err={}'.format(*return_tuple)
+def assert_rc(actual, expected, assert_report):
+    assert actual == expected, assert_report
+
+
+def exec_cmd_report(exec_cmd_output):
```

Comment:
I believe you once said that methods should be verbs and I think that report does not really match here. Also `exec_cmd_output` could maybe be better `exec_cmd_result`. With a verb it could be:

`format_exec_cmd_result(result)`

Reply:
I think I would even rename `exec_cmd` to `exec` and make it return an object with a `__str__()` method that does what `exex_cmd_report` does.

Reply:
I think it is an overkill to objectize this. We would need to create an object and then execute it (execution as part of the constructor is not a preferred approach).

Anyway, generating a report from the output seems reasonable to me.

Regarding the verb, I considered the `exec` prefix as such, but I'm ok with your suggestion.

Reply:
I would also be fine with just having a result object but it looks good enough for me now as well.